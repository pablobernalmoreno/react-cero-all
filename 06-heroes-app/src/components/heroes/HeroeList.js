import React, { useMemo } from "react";
import { getHeroesByPublisher } from "../../selectors/getHeroesByPublisher";
import { HeroCard } from "./HeroCard";

export const HeroList = ({ publisher }) => {
  const heroes = useMemo(() => getHeroesByPublisher(publisher), [publisher]);

  return (
    <div className="card-columns">
      {heroes.map((hero) => (
        <HeroCard
          key={hero.id}
          id={hero.id}
          superheroe={hero.superhero}
          publisher={hero.publisher}
          alter_ego={hero.alter_ego}
          first_appearance={hero.first_appearance}
          characters={hero.characters}
        />
      ))}
    </div>
  );
};
