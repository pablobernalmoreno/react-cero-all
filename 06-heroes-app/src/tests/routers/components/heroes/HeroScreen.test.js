import { mount } from "enzyme";
import { MemoryRouter, Route } from "react-router-dom";
import { HeroScreen } from "../../../../components/heroes/HeroScreen";

describe("Pruebas en el componente <HeroScreen />", () => {
  const history = {
    length: 10,
    push: jest.fn(),
    goBack: jest.fn(),
  };

  const wrapper = mount(
    <MemoryRouter initialEntries={["/hero"]}>
      <HeroScreen history={history} />
    </MemoryRouter>
  );

  test("debe de mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de mostrar un heroe si el parÃ¡metro existe y se encuentra", () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={["/hero/Batman"]}>
        <Route path="/hero/:heroeId" component={HeroScreen} />
      </MemoryRouter>
    );

    expect(wrapper.find(".row").exists()).toBeTruthy();
  });

  test("debe de regresar a la pantalla anterior con Push", () => {
    const history = {
      length: 1,
      push: jest.fn(),
      goBack: jest.fn(),
    };

    const wrapper = mount(
      <MemoryRouter initialEntries={["/hero/Batman"]}>
        <Route
          path="/hero/:heroeId"
          component={(props) => <HeroScreen history={history} />}
        />
      </MemoryRouter>
    );

    wrapper.find("button").simulate("click");

    expect(history.push).toBeCalledWith("/");

    expect(history.goBack).not.toBeCalledWith();
  });

  test("debe de regresar a la pantalla anterior con goBack", () => {
    const history = {
      length: 10,
      push: jest.fn(),
      goBack: jest.fn(),
    };

    const wrapper = mount(
      <MemoryRouter initialEntries={["/hero/Batman"]}>
        <Route
          path="/hero/:heroeId"
          component={(props) => <HeroScreen history={history} />}
        />
      </MemoryRouter>
    );

    wrapper.find("button").simulate("click");

    expect(history.push).not.toBeCalledWith("/");

    expect(history.goBack).toBeCalledWith();
  });

  test("debe de llamar el redirect si el 'hero' no existe", () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={["/hero/Batman12312412"]}>
        <Route
          path="/hero/:heroeId"
          component={(props) => <HeroScreen history={history} />}
        />
      </MemoryRouter>
    );

    expect(wrapper.text()).toBe(
      "    Alter ego:     Publisher:     First appearance:    Characters   Return"
    );
  });
});
