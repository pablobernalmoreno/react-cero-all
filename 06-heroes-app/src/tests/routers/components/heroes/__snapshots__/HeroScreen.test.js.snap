// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <HeroScreen /> debe de mostrarse correctamente 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      "/hero",
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "a4cwrh",
            "pathname": "/hero",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "a4cwrh",
          "pathname": "/hero",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <HeroScreen
      history={
        Object {
          "goBack": [MockFunction],
          "length": 10,
          "push": [MockFunction],
        }
      }
    >
      <div
        className="row mt-5"
      >
        <div
          className="col-4"
        >
          <img
            className="img-thumbnail animate__animated animate__fadeInLeft"
            src="../assets/heroes/undefined.jpg"
          />
        </div>
        <div
          className="col-8 animate__animated animate__fadeIn"
        >
          <h3>
             
             
          </h3>
          <ul
            className="list-group list-group-flush"
          >
            <li
              className="list-group-item"
            >
               
              <b>
                 Alter ego: 
              </b>
               
               
            </li>
            <li
              className="list-group-item"
            >
               
              <b>
                 Publisher: 
              </b>
               
               
            </li>
            <li
              className="list-group-item"
            >
               
              <b>
                 First appearance: 
              </b>
               
               
            </li>
          </ul>
          <h5>
             Characters 
          </h5>
          <p>
             
             
          </p>
          <button
            className="btn btn-outline-info"
            onClick={[Function]}
          >
            Return
          </button>
        </div>
      </div>
    </HeroScreen>
  </Router>
</MemoryRouter>
`;
